interface/Tools.h

```diff
@@ -74,6 +76,261 @@ MELAAngles HHAnalyzer::getMELAAngles(const LorentzVector &q1, const LorentzVecto
     return angles;
 }
 
+void HHAnalyzer::matchOfflineLepton(const HLTProducer& hlt, HH::Dilepton& dilepton) {
+// Function that tries to match `lepton` with an online object, using a deltaR and a deltaPt cut
+// Returns the index inside the HLTProducer collection, or -1 if no match is found.
+// (Originally taken from https://github.com/blinkseb/TTAnalysis/blob/c2a2d5de3e4281943c19c582afb452b8ef6457f1/plugins/TTAnalyzer.cc#L533)
+    if (leptons[dilepton.ilep1].hlt_already_tried_matching && leptons[dilepton.ilep2].hlt_already_tried_matching) {
```

Comment:
Could'nt we hit a situation where the two considered leptons have already been through the process of matching but separated from each other? If this is the case and that each time we did not find a common path, line 174 and 178 would make us skip the matching process for a pair that did not go together through it?